### DISCUSSION TITLE
Attenuation correction is not as expected.

### DISCUSSION THREAD

**Discussion by SomaKohsei** [villekf/OMEGA#40] [Q&A]
Hi, 
I am enjoying learning this software.

I did a GATE simulation of imaging a NEMA phantom using a PET system (Discovery ST).
Without attenuation correction (AC) , the results are as expected. (higher counts around the phantom).
However, with AC, I got an unexpected images.
This leads us to believe that the AC image may be rotated 90 degrees.
The input MuMap.mhd/raw is not rotated 90 degrees when displayed in Imagej or vv.
What is wrong? I tried changing the TransformMatrix in MuMap.mhd in various ways but could not solve the problem.
The AC image was obtained from MuMap_Actor in GATE.
![question](https://user-images.githubusercontent.com/35340150/231168903-2e02417c-0547-490d-b647-e9302a8bf936.JPG)

The reconstructed image also has 100 million pixel counts, which is very large.

Thank you.
Soma

**Comment by villekf** [comment]
The MuMap image is indeed rotated by 90 degrees. This was because, in my test cases, the MuMap image had to be rotated by 90 degrees to match the reconstruction. However, one thing I hadn't taken into account was that you can also rotate the PET reconstruction which in turn can mess up the correct orientation of the attenuation image completely. I guess I need to have a separate rotation variable for the attenuation image itself. In the meantime, however, you can modify (or simply comment) this [line](https://github.com/villekf/OMEGA/blob/master/source/set_up_corrections.m#L95) in order to get the correct rotation. Another alternative is to use `options.offangle` variable to rotate the PET data to match the attenuation image. At the moment, the `TransformMatrix` variable is not taken into account at all.

I'll try to get this properly fixed in the next release.

**Comment by SomaKohsei** [comment]
Thank you for your reply :)

I commented out the "rot90" line, and now the PET image and MuMap face 180° in opposite directions.
Now I haven't solved the problem yet, but it was a clue.
I'll look into it some more and see what I can come up with.
If I can't solve the problem, please let me ask again.

Regards,

**Comment by SomaKohsei** [comment]
The problem I asked about last time is still unresolved.
Commenting out the rot90( ) line in set_up_correction.m and changing the number of rotations did not give the correct result.

Creating a MuMap in GATE with the MuMap flipped upside down did not work.

So, could you please tell me a little more about options.offangle?
![question3](https://user-images.githubusercontent.com/35340150/232736979-fe206c94-d403-40aa-9f94-07f1fb2967d6.JPG)

The following changes did not change the results.

(gate_main.m)
options.flip_image = false/true
options.offangle = options.det_w_pseudo * (X/4);     X=1,2,3 ....

**Reply by villekf** [reply]
Have you visualized the MuMap image in Matlab/Octave to see the orientation there? The `options.vaimennus` contains the attenuation image. You should aim for it to have the same orientation as the reconstructed image. Another thing is, that apparently is missing in `gate_main.m`, are the units in 1/cm?

**Reply by SomaKohsei** [reply]
------Another thing is, that apparently is missing in gate_main.m, are the units in 1/cm?
Do I just need to prepare a MuMap in units of 1/cm, right?

------Have you visualized the MuMap image in Matlab/Octave to see the orientation there? The options.vaimennus contains the attenuation image. You should aim for it to have the same orientation as the reconstructed image.
I am not familiar with Octave, so it may be difficult for me to do that. I will study it a bit and try.

thank you.

**Reply by villekf** [reply]
Either do it in GATE with 1/cm or just multiply/divide the values to get 1/cm, depending on the original units.
